types.go

```diff
@@ -42,21 +48,52 @@ func (r Reference) ByteSize() int {
 	return len(r)
 }
 
-func toBytes(data interface{}) ([]byte, error) {
-	switch castData := data.(type) {
-	case []uint8:
-		return castData, nil
-	case uint32:
-		var buf [4]byte
-		binary.BigEndian.PutUint32(buf[:], castData)
-		return buf[:], nil
+// Scalar represents a JSON or JSON-LD scalar (string, number, true or false)
+type Scalar struct {
+	value interface{}
+}
+
+// ErrInvalidValue is returned when an invalid value is parsed
+var ErrInvalidValue = errors.New("invalid value")
```

Comment:
Unsupported value.

Maybe add provided type? (like `fmt.Errorf("unsupported value: %T", value)`)

Reply:
that would be for the caller